packages/react-rest/src/services/useOne/useOne.ts

```diff
@@ -10,10 +10,11 @@ import type {
   Source,
 } from "@hatchifyjs/rest-client"
 
-const useMemoizedQuery = (query: QueryOne) => {
+const useMemoizedQuery = (query: QueryOne | string) => {
   return useMemo(() => {
-    return query
-  }, [query.include, query.fields, query.id])
+    const queryObj = typeof query === "string" ? { id: query } : { ...query }
+    return queryObj
+  }, [query])
```

Comment:
does this deps array cause that infinite loop problem we had in `useAll` if the query that is passed in is contains a `fields` object?

if we don't have a test for it, can we write one that tests `useOne` hook and passes in something like:
```
useOne({ id: "10", fields: {} })
```